#!/usr/bin/env python

PACKAGE = "singaboat_vrx"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("task_1_launch_path", str_t, 0, "Path to station-keeping task launch file", "/launch/submissions/vrx_station_keeping.launch")
gen.add("task_2_launch_path", str_t, 0, "Path to wayfinding task launch file", "/launch/submissions/vrx_wayfinding.launch")
gen.add("task_3_launch_path", str_t, 0, "Path to scene perception task launch file", "/launch/submissions/vrx_scene_perception.launch")
gen.add("task_4_launch_path", str_t, 0, "Path to semantic navigation task launch file", "/launch/submissions/vrx_semantic_navigation.launch")
gen.add("task_5_launch_path", str_t, 0, "Path to gymkhana challenge task launch file", "/launch/submissions/vrx_gymkhana_challenge.launch")
gen.add("task_6_launch_path", str_t, 0, "Path to scan-dock-deliver task launch file", "/launch/submissions/vrx_scan_dock_deliver.launch")

gen.add("debug", bool_t, 0, "Flag to enable/disable debug messages", True)

exit(gen.generate(PACKAGE, "singaboat_vrx", "TaskManager"))
